kafka-event-listener/src/main/java/org/hypertrace/core/kafka/event/listener/KafkaLiveEventListener.java

```diff
@@ -84,10 +85,26 @@ public Builder<K, V> withExecutorService(
 
     public KafkaLiveEventListener<K, V> build(
         String consumerName, Config kafkaConfig, Consumer<K, V> kafkaConsumer) {
+      if (executorService == null) {
+        executorService =
+            Executors.newSingleThreadExecutor(new ListenerThreadFactory(consumerName));
+      }
       return new KafkaLiveEventListener<>(
           new KafkaLiveEventListenerCallable<>(consumerName, kafkaConfig, kafkaConsumer, callbacks),
           executorService,
           cleanupExecutor);
     }
   }
 }
+
+class ListenerThreadFactory implements ThreadFactory {
+  private final String name;
+
+  public ListenerThreadFactory(String consumerName) {
+    this.name = "kafka-live-event-listener-" + consumerName + "-" + System.currentTimeMillis();
```

Comment:
```suggestion
    this.name = "kafka-live-event-listener-" + consumerName + "-" + String.valueOf(System.currentTimeMillis());
```

Reply:
<img width="292" alt="image" src="https://github.com/hypertrace/kafka-streams-framework/assets/40003481/7efe8557-0a48-46e4-98aa-5dd2f2d251ec">
it is not needed